src/builder_state.rs

```diff
@@ -692,7 +705,7 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
                 requested_view_number,
             ));
 
-            let encoded_txns: Vec<u8> = payload.encode().unwrap().to_vec();
+            let encoded_txns: Vec<u8> = payload.encode().ok()?.to_vec();
```

Comment:
We should log something if we can't encode the payload

Reply:
I think that's a can't fail for the actual Payload, but, yeah, the trait makes no such guarantee.

Reply:
Fixed in [f1322c5](https://github.com/EspressoSystems/hotshot-builder-core/pull/125/commits/f1322c50ba053956be034172b621ab0ab4ce3704)